lib/datadog/grok/src/field_traversal.rs

```diff
@@ -0,0 +1,120 @@
+use lookup::{Field, FieldBuf, Lookup, LookupBuf, Segment, SegmentBuf};
+use parsing::value::Value;
+use snafu::Snafu;
+use std::collections::BTreeMap;
+use std::fmt::Debug;
+
+pub fn get_field<'a>(
+    source: &Value,
+    lookup: impl Into<Lookup<'a>> + Debug,
+) -> std::result::Result<Option<&Value>, Error> {
+    let mut working_lookup = lookup.into();
+
+    let this_segment = working_lookup.pop_front();
+    match (this_segment, source) {
+        // We've met an end and found our value.
+        (None, item) => Ok(Some(item)),
+        // Descend into a map
+        (Some(Segment::Field(Field { name, .. })), Value::Map(map)) => match map.get(name) {
+            Some(inner) => get_field(inner, working_lookup.clone()),
```

Comment:
```suggestion
            Some(inner) => get_field(inner, working_lookup),
```

I think you can drop this clone.